### PR TITLE
Simplify code and card detection

### PR DESCRIPTION
**Pull request by metsma**
Signed-off-by: Raul Metsma <raul@metsma.ee>

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [X] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: src/libopensc/card-mcrd.c [hunk 1]
```diff
@@ -26,15 +26,15 @@
...
-
 #include "internal.h"
 #include "asn1.h"
 #include "cardctl.h"
 #include "gp.h"
 
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+
```
Comment (Jakuje) [inline src/libopensc/card-mcrd.c]: Why is this moved below? It is a good practice to have system includes on the top and local on the bottom, as it is in the rest of the code base.

Comment (metsma) [reply src/libopensc/card-mcrd.c]: Reverted,  but I follow the C++ guidlines:
* corresponded header file
* necessary project headers
* 3rd party libraries headers
* standard libraries headers
* system headers
